File: src/Algorithms/Greedy/medium/MaxMin.java
//Date 25-9-2018

package Algorithms.Greedy.medium;

import java.util.Arrays;
import java.util.Scanner;

public class MaxMin {

    public static void main(String [] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int k = in.nextInt();
        int [] ar = new int[n];
        for(int i=0;i<n;i++)
            ar[i] = in.nextInt();

        Arrays.sort(ar);
        int min = ar[k-1] - ar[0];
        int current;
        for (int i=1;i<n-k+1;i++) {
            current = ar[i+k-1] - ar[i];
            if(current<min)
            {
                min = current;
                if(min==0) break;
            }
        }
        System.out.println(min);
        in.close();
    }
}

File: src/TheHackerRankInterviewPreparationKit/Greedy_Algorithms/Max Min.kt
package TheHackerRankInterviewPreparationKit.Greedy_Algorithms
//Date 12-8-2018
import java.util.*

fun main(args : Array<String>){

    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val k = s.nextInt()
    val arr = IntArray(n)

    for( i in 0 until n)
        arr[i] = s.nextInt()

    Arrays.sort(arr)

    var min = arr[k-1]-arr[0]
    var current: Int

    for( i in 1 until n-k+1)
    {
        current = arr[i+k-1]-arr[i]
        if( current < min)
        {
            min = current
            if( min == 0)
                break
        }
    }
    println(min)
    s.close()
}